metabase/metabase
# Date/time column extractions

Closes #39248 

Depends on #39162

### Description

Tests will be added later, when BE is ready: #39291

3 e2e test suites are failing, but it's because of target PR (#39162).

#### Not everything is ready in the backend
- the extraction drill is shown for table cells (should be shown only for table headers)
- the order of extraction types is reversed (should go from "Hour of day" to "Year")
- using extraction twice will result in duplicated column name error
- year is incorrectly formatted (should be `2025` not, `2,025`)
- months should be shown as names, not numbers
- quarters should be shown as Q1, Q2 etc., not numbers
- "Day" should be called "Day of the month"
- "Hour of day" and "Day of week" extractions are missing
- "Week" extraction should not be present
- etc.

### How to verify

1. Create a question with a date/time column, and visualize it as table
2. Click header of the date/time column
3. Click "Extract day, month..."
4. Choose extraction type
5. New column should appear in the table and in notebook editor

### Demo

https://github.com/metabase/metabase/assets/6830683/ceaeb7d9-20a6-4935-8f5d-f6f9f8f26abf
